Add support for Python 3.11 and Django 4.1

roaldnefs opened the pull request:
Add support for Python 3.11 and Django 4.1.

roaldnefs committed:
feat: add support for Python 3.11 and Django 4.1

Signed-off-by: Roald Nefs <info@roaldnefs.com>

roaldnefs committed:
fix: skip running tests on Django version 4.1 and PostgreSQL 10

Signed-off-by: Roald Nefs <info@roaldnefs.com>

roaldnefs committed:
fix: skip running tests on Django version 4.1 and Python 3.7

Signed-off-by: Roald Nefs <info@roaldnefs.com>

matthijswendelaar commented on .github/workflows/test.yaml:
```diff
@@ -10,14 +10,19 @@ jobs:
             django-version: 4.0
+          - postgres-version: 10
+            django-version: 4.1
+          # Django 4.1 requires Python >=3.8
```
A comment has been added here to describe why this django-version is not compatible with python 3.7. Is this any different from the other excludes that are listed here?

If not, I would suggest to also add comments for the other excludes describing why they are excluded for testing.

roaldnefs replied:
I've added the comments in ed92632123b6651929a139132422b3669dac0e1e as this isn't any different.

roaldnefs committed:
docs(ci): describe excluded tests in GH actions

Describe the excluded tests in the GitHub action for running the tests
as suggested by Matthijs Wendelaar in
https://github.com/warpnet/django-nacl-fields/pull/48#discussion_r1062274555.

Signed-off-by: Roald Nefs <info@roaldnefs.com>
